docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,296 @@
+   for YAML files, and no other files are examined for that target
+   path;
 
-Flux can be configured to confine the scope of (1) and (2):
+ - if no `.flux.yaml` file is found, the usual behaviour of looking
+   for YAML files is adopted for that target path.
 
-* To specific (sub)directories (flag `--git-path`)
-* To a Git branch other than `master` (flag `--git-branch`)
+The manifests from all the target paths -- read from YAML files or
+generated -- are combined before applying to the cluster. If
+duplicates are detected, an error is logged and fluxd will abandon the
+attempt to apply manifests to the cluster.
 
-## Abstracting out file-access: generators and updaters
+Here are some examples:
 
-Flux allows you to declare configuration files to override file operations (1) and (2), by declaring commands which
-perform equivalent actions.
+```
+.
+├── base
+│   ├── demo-ns.yaml
+│   ├── kustomization.yaml
+│   ├── podinfo-dep.yaml
+│   ├── podinfo-hpa.yaml
+│   └── podinfo-svc.yaml
+├── .flux.yaml
+├── production
+│   ├── flux-patch.yaml
+│   ├── kustomization.yaml
+│   └── replicas-patch.yaml
+└── staging
+    ├── flux-patch.yaml
+    └── kustomization.yaml
+```
 
-The configuration files are formatted in `YAML` and named `.flux.yaml`. They must be located on the Git repository
-(more on this later).
+In this case, say you started `fluxd` with `--git-path=staging`, it
+would find `.flux.yaml` in the top directory and use that to generate
+manifests.  The other files and directories (if there were any) in
+`staging/` are not examined by fluxd, in favour of following the
+instructions given in the `.flux.yaml` file.
 
-A `commandUpdated` `.flux.yaml` file has the following loosely specified format:
+This layout could also be used with `--git-path=production`.
+
+In this modified example, the `.flux.yaml` file has been moved under
+`staging/`:
 
-```yaml
-version: X
-commandUpdated:
-  generators:
-    - command: generator_command1 g1arg1 g1arg2 ...
-    - command: generator_command2 g2arg2 g2arg2 ...
-  updaters:
-    - containerImage:
-        command: containerImage_updater_command1 ciu1arg1 ciu1arg2 ...
-      policy:
-        command: policy_updater_command1 pu1arg1 pu1arg2 ...
-    - containerImage:
-        command: containerImage_updater_command2 ciu2arg1 ciu2arg2 ...
-      policy:
-        command: policy_updater_command2 pu2arg1 pu2arg2 ...
 ```
+.
+├── base
+│   ├── demo-ns.yaml
+│   ├── kustomization.yaml
+│   ├── podinfo-dep.yaml
+│   ├── podinfo-hpa.yaml
+│   └── podinfo-svc.yaml
+├── production
+│   ├── flux-patch.yaml
+│   ├── kustomization.yaml
+│   └── replicas-patch.yaml
+└── staging
+    ├── flux-patch.yaml
+    ├── .flux.yaml
+    └── kustomization.yaml
+```
+
+… since the `.flux.yaml` file is now under `staging/`, it will still
+take effect for `--git-path=staging`. However:
+
+Using `--git-path=production` would **not produce a usable
+configuration**, because without an applicable `.flux.yaml`, the files
```

Comment:
Even if redundant, I think it's worth indicating that a `.flux.yaml` file cannot be found in that case because there is non in `/production` or its parent directories (`/`)